pywwt/layers.py

```diff
@@ -236,40 +256,72 @@ class TableLayer(HasTraits):
+                      '(`float`)', allow_none=True).tag(wwt=None)
 
     # NOTE: we deliberately don't link cmap_att to colorMapColumn because we
-    # need to compute the colors ourselves based on the min/max and then use the
-    # resulting column.
-    cmap_att = Unicode(help='The column to use for the colormap')
-    cmap_vmin = Float(None, allow_none=True)
-    cmap_vmax = Float(None, allow_none=True)
-    cmap = Any(cm.viridis, help='The Matplotlib colormap')
-
-    color = Color('white', help='The color of the markers').tag(wwt='color')
-    opacity = Float(1, help='The opacity of the markers').tag(wwt='opacity')
-
-    marker_type = Unicode('gaussian', help='The type of marker').tag(wwt='plotType')
-    marker_scale = Unicode('screen', help='Whether the scale is defined in '
-                           'world or pixel coordinates').tag(wwt='markerScale')
-
-    far_side_visible = Bool(False, help='Whether markers on the far side are '
-                            'visible').tag(wwt='showFarSide')
+    # need to compute the colors ourselves based on the min/max and then use
+    # the resulting column.
+    cmap_att = Unicode(help='The column to use for the colormap '
+                       '(`str`)').tag(wwt=None)
+    cmap_vmin = Float(None, help='The minimum level of the colormap. Found '
+                      'automagically once cmap_att is set (`float`)',
+                      allow_none=True).tag(wwt=None)
+    cmap_vmax = Float(None, help='The maximum level of the colormap. Found '
+                      'automagically once cmap_att is set (`float`)',
+                      allow_none=True).tag(wwt=None)
+    cmap = Any(cm.magma, help='The Matplotlib colormap '
+               '(:class:`matplotlib.colors.ListedColormap`)').tag(wwt=None)
+
+    color = Color('white', help='The color of the markers '
+                  '(`str` or `tuple`)').tag(wwt='color')
+    opacity = Float(1, help='The opacity of the markers '
+                    '(`str`)').tag(wwt='opacity')
+
+    marker_type = Unicode('gaussian', help='The type of marker '
+                          '(`str`)').tag(wwt='plotType')
+    marker_scale = Unicode('screen', help='Whether the scale is '
+                           'defined in world or pixel coordinates '
+                           '(`str`)').tag(wwt='markerScale')
+
+    far_side_visible = Bool(False, help='Whether markers on the far side '
+                            'of a 3D object are visible '
+                            '(`bool`)').tag(wwt='showFarSide')
+
+    # NOTE: we deliberately don't link time_att to startDateColumn here
+    # because we need to compute a new times column based on time_att before
+    # passing the result on to WWT
+    time_att = Unicode(help='The column to use for time (`str`)').tag(wwt=None)
+    time_series = Bool(False, help='Whether the layer contains time series '
+                       'elements (`bool`)').tag(wwt='timeSeries')
```

Comment:
Do we need this separate option, or would it be sufficient to check that ``time_att`` is not None? (there's a way to get a Unicode trait to also accept None, check the traitlets docs as I have forgotten the exact option name)

Reply:
@pkgw and I were also talking about this. I don't have a strong opinion on it. My main goal was to try to replicate the way layer creation works in the code from the SDK, but we naturally have the latitude to change this in `pywwt` if we desire.

Reply:
As I think about it, I think I might actually prefer the idea of keeping the separate boolean. *If* we add code that guesses the time column automatically, it could be pretty confusing to add a layer and not see anything appear because it was automatically assumed to be time-dependent. On the other hand, overriding with a `time_att = None` wouldn't be so hard.